cli/flox/src/commands/mod.rs

```diff
@@ -516,25 +516,39 @@ impl UpdateNotification {
             Err(e) => Err(UpdateNotificationError::Io(e))?,
         };
 
-        let new_version = get_latest_version_future.await?;
+        let new_version_str = get_latest_version_future.await?;
+        let current_version_str = FLOX_VERSION.as_str();
 
-        // Sanity check we got a version back
-        if let Err(e) = semver::Version::parse(&new_version) {
-            return Err(UpdateNotificationError::WeMayHaveMessedUp(anyhow!(
-                "version is invalid: {e}"
-            )));
+        if let Ok(new_version) = new_version_str.parse::<FloxVersion>() {
+            if let Ok(current_version) = current_version_str.parse::<FloxVersion>() {
+                match current_version.partial_cmp(&new_version) {
+                    None => {
```

Comment:
nonblocking: we report errors to sentry, and I think this will give us an error if someone checks for an update when they have a flake style version. Not the end of the world, but would be nice to avoid

Reply:
Did you refactoring already addressed this?

Reply:
No we still have:
```
            None => Err(UpdateNotificationError::WeMayHaveMessedUp(anyhow!(
                "Cannot compare the current version {current_version} to {new_version}"
            ))),
```